src/megaclient.cpp

```diff
@@ -7878,15 +7879,16 @@ void MegaClient::applykeys()
 {
     CodeCounter::ScopeTimer ccst(performanceStats.applyKeys);
 
-    int t = 0;
-
     // FIXME: rather than iterating through the whole node set, maintain subset
     // with missing keys
-    for (node_map::iterator it = nodes.begin(); it != nodes.end(); it++)
+
+    int noKeyExpected = (rootnodes[0] != UNDEF) + (rootnodes[1] != UNDEF) + (rootnodes[2] != UNDEF);
+
+    if (nodes.size() > size_t(mAppliedKeyNodeCount + noKeyExpected))
     {
-        if (it->second->applykey())
+        for (auto& it : nodes)
         {
-            t++;
+            it.second->applykey();
```

Comment:
We might want to have the if within the for. Hence, if mAppliedKeyNodeCount reaches the number of appliable keys, we break the loop.

Reply:
Thanks for the suggestion - I think in this case it's cleaner and clearer to have the check outside - we do run this loop all the time currently, so it's not slow, it's just that we do it very frequently, which this PR should address.    In the very rare case where we could exit the loop early, there's nothing to do for the rest of the keys so each applykey() call will return immediately, and I'm not sure that would save more time than the extra check per iteration would cost anyway. thanks!